storage/watchable_store_test.go

```diff
@@ -59,6 +59,134 @@ func TestNewWatcherCancel(t *testing.T) {
 	}
 }
 
+// TestCancelUnsynced tests if running CancelFunc removes watchings from
+// synced and unsynced.
```

Comment:
i failed not see how it ensures removing watchings from synced. I think this test only ensures latter?